Tests/SwiftIfConfigTest/EvaluateTests.swift

```diff
@@ -0,0 +1,149 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import XCTest
+import SwiftSyntax
+import SwiftParser
+import SwiftIfConfig
+import _SwiftSyntaxTestSupport
+
+public class EvaluateTests: XCTestCase {
+  func testLiterals() throws {
+    let buildConfig = TestingBuildConfiguration(customConditions: ["DEBUG", "ASSERTS"])
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("true"), .active)
+    XCTAssertEqual(try ifConfigState("false"), .inactive)
+
+    // FIXME: How can we produce warnings from this code?
+    XCTAssertEqual(try ifConfigState("1"), .active)
+    XCTAssertEqual(try ifConfigState("0"), .inactive)
+  }
+
+  func testCustomConfigs() throws {
+    let buildConfig = TestingBuildConfiguration(customConditions: ["DEBUG", "ASSERTS"])
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertEqual(try ifConfigState("DEBUG"), .active)
+    XCTAssertEqual(try ifConfigState("NODEBUG"), .inactive)
+    XCTAssertEqual(try ifConfigState("!DEBUG"), .inactive)
+    XCTAssertEqual(try ifConfigState("!NODEBUG"), .active)
+    XCTAssertEqual(try ifConfigState("DEBUG && ASSERTS"), .active)
+    XCTAssertEqual(try ifConfigState("DEBUG && nope"), .inactive)
+    XCTAssertEqual(try ifConfigState("nope && DEBUG"), .inactive)
+    XCTAssertEqual(try ifConfigState("nope && 3.14159"), .inactive)
+    XCTAssertEqual(try ifConfigState("DEBUG || ASSERTS"), .active)
+    XCTAssertEqual(try ifConfigState("DEBUG || nope"), .active)
+    XCTAssertEqual(try ifConfigState("nope || DEBUG"), .active)
+    XCTAssertEqual(try ifConfigState("nope || !DEBUG"), .inactive)
+    XCTAssertEqual(try ifConfigState("DEBUG || 3.14159"), .active)
+    XCTAssertEqual(try ifConfigState("(DEBUG) || 3.14159"), .active)
+  }
+
+  func testBadExpressions() throws {
+    let buildConfig = TestingBuildConfiguration(customConditions: ["DEBUG", "ASSERTS"])
+
+    func ifConfigState(_ condition: ExprSyntax) throws -> IfConfigState {
+      try IfConfigState(condition: condition, configuration: buildConfig)
+    }
+
+    XCTAssertThrowsError(try ifConfigState("3.14159")) { error in
+      XCTAssertEqual(String(describing: error), "invalid conditional compilation expression")
+    }
```

Comment:
If `#if` condition evaluation produces diagnostics, it would probably be good to have a `assertIfConfigEvaluation` that accepts a `DiagnosticSpec` and location markers.

And honestly, I think it would be a good idea even in the current design to avoid the repeated definition of `ifConfigState`.

Reply:
This is done now